scripts/TestHarness/testers/RAVENImageDiff.py

```diff
@@ -56,60 +59,109 @@ def diff(self):
+        return (False, self.self.__message)
       try:
         # RAK - The original line...
         # gold_image = imread(open(gold_filename,'r'))
         # ...didn't work on Windows Python because it couldn't sense the file type
         gold_image = imread(gold_filename)
       except IOError:
         files_read = False
-        self.__messages += 'Unrecognized file type for test image in scipy.imread: '+gold_filename
-        return (False, self.__messages)
+        self.self.__message += 'Unrecognized file type for test image in scipy.imread: '+gold_filename
+        return (False, self.self.__message)
       #first check dimensionality
       if gold_image.shape != test_image.shape:
-        self.__messages += 'Gold and test image are not the same shape: '+\
+        self.self.__message += 'Gold and test image are not the same shape: '+\
           str(gold_image.shape)+', '+str(test_image.shape)
         self.__same = False
-        return (self.__same, self.__messages)
-      #set default options
-      DU.set_default_options(self.__options)
+        return (self.__same, self.self.__message)
       #pixelwise comparison
       #TODO in the future we can add greyscale, normalized coloring, etc.
       # For now just do raw comparison of right/wrong pixels
       diff = gold_image - test_image
-      only_diffs = diff[abs(diff) > self.__options['zero_threshold']]
+      only_diffs = diff[abs(diff) > self.__zero_threshold]
       pct_num_diff = only_diffs.size/float(diff.size)
-      if pct_num_diff > self.__options['rel_err']:
-        self.__messages += 'Difference between images is too large:'+\
+      if pct_num_diff > self.__rel_err:
+        self.self.__message += 'Difference between images is too large:'+\
           ' %2.2f pct (allowable: %2.2f)' %(100*pct_num_diff,
-                                            100*self.__options['rel_err'])
+                                            100*self.__rel_err)
         self.__same = False
-    return (self.__same, self.__messages)
+    return (self.__same, self.__message)
+
+
+class ImageDiffer(Differ):
+  """
+  This is the class to use for handling the parameters block.
+  """
+
+  @staticmethod
+  def get_valid_params():
+    """
+      Returns the valid parameters for this class.
+      @ In, None
+      @ Out, params, _ValidParameters, return the parameters.
+    """
+    params = Differ.get_valid_params()
+    params.add_param('rel_err', '', 'Relative Error for image files')
+    params.add_param('zero_threshold', sys.float_info.min*4.0,
+                     'it represents the value below which a float is '+
+                     'considered zero in the pixel comparison')
+    return params
+
+  def __init__(self, name, params, test_dir):
+    """
+      Initializer for the class. Takes a String name and a dictionary params
+      @ In, name, string, name of the test.
+      @ In, params, dictionary, parameters for the class
+      @ In, test_dir, string, path to the test.
```

Comment:
Camelback?

Reply:
The rest of rook is PEP8 instead of camelBack so I would consider this acceptable.